pkg/tfgen/generate_nodejs_utilities.go

```diff
@@ -47,4 +49,30 @@ export function requireWithDefault<T>(req: () => T, def: T | undefined): T {
     }
     return def;
 }
+
+export function unwrap(val: pulumi.Input<any>): pulumi.Input<any> {
+    // Bottom out at primitives.
+    if (val === undefined || typeof val !== 'object') {
+        return val;
+    }
+
+    // Recurse on outputs, promises, arrays, and objects.
+    if (pulumi.Output.isInstance(val)) {
+        return val.apply(unwrap);
+    }
+    if (val instanceof Promise || val instanceof Array) {
+        return pulumi.output(val).apply(unwrap);
```

Comment:
i'm surprised this works for Array. Something doesn't feel right htere.  I would think we'd first need to call .map(unwrap) on the array elements, then do a pulumi.all on the result of that.

Reply:
It doesn't--we do need `map`. This was a bad edit (side note: I wish we could use an actual TS file here).